asv_bench/benchmarks/io_bench.py

```diff
@@ -192,3 +192,25 @@ def time_read_nrows(self, compression, engine):
             ext = ".bz2"
         pd.read_csv(self.big_fname + ext, nrows=10,
                     compression=compression, engine=engine)
+
+
+class read_json_lines(object):
+    goal_time = 0.2
+
+    def setup(self):
+        self.N = 100000
+        self.C = 5
+        self.df = DataFrame(dict([('float{0}'.format(i), randn(self.N)) for i in range(self.C)]))
+        self.df.to_json("__test__.json",orient="records",lines=True)
```

Comment:
add a ``teardown`` method to remove the created file here (you can name it like)

```
class read_json_lines(...):
      file = '__test__.json'

      def teardown(self):
             try:
                  os.remove(self.file)
             except:
                  pass
```

Reply:
I think we need to scour the asv's for not-using teardowns as well (almost all prob in io benchmarks). @louispotok if you could add would be great (or could do in another issue)

Reply:
Created a new issue [here](https://github.com/pandas-dev/pandas/issues/17616), I'm happy to do it.

Reply:
And, created the teardown method.